top.mk

```diff
@@ -232,9 +235,13 @@ OPTEE_EXTRA += CFG_CORE_HEAP_SIZE=524288
 #OPTEE_EXTRA += CFG_TA_ASLR=n
 
 # Tell TFA where to find the OP-TEE binaries
-TFA_EXTRA += BL32=$(OPTEE_OUTPUT)/arm-plat-vexpress/core/tee-header_v2.bin
-TFA_EXTRA += BL32_EXTRA1=$(OPTEE_OUTPUT)/arm-plat-vexpress/core/tee-pager_v2.bin
-TFA_EXTRA += BL32_EXTRA2=$(OPTEE_OUTPUT)/arm-plat-vexpress/core/tee-pageable_v2.bin
+ifneq ($(USE_TEE_BIN),)
+TFA_EXTRA += BL32=$(OPTEE_OUTPUT)/arm-plat-$(OPTEE_PLATFORM)/core/tee.bin
+else
```

Comment:
In this case of USE_TEE_BIN, add a new target for generating tee.bin that depends on the optee_os target, and then add a dependency to the TFA target to make sure it gets built first. Adding the generation of tee.bin to the optee_os/% target will break things like 'make optee_os/clean'

Reply:
ag good point. That will get rid of the ugly if in the makefile.

Reply:
Looking at this again, it should work as is even when cleaning targets. FIP_DEPS += optee_os/all is already defined in the makefile when OPTEE is selected.
In any case, I'll just make it a different target